src/components/Tutorial/__tests__/Tutorial.ts

```diff
@@ -0,0 +1,317 @@
+
+beforeEach(createTestAppWithTutorial)
+afterEach(cleanupTestApp)
+
+describe('Tutorial 1', async () => {
+  describe('step start', () => {
+    it('shows the welcome text', () => {
+      expect(screen.getByText('Welcome to your personal thoughtspace.')).toBeInTheDocument()
+    })
+
+    it('shows the "Next" button, that leads to the first step', async () => {
+      const user = userEvent.setup({ delay: null })
+      await user.click(screen.getByText('Next'))
+      await act(vi.runOnlyPendingTimersAsync)
+      expect(
+        screen.getByText('First, let me show you how to create a new thought', { exact: false }),
+      ).toBeInTheDocument()
+    })
+  })
+
+  it('step first thought - how to create a thought ', async () => {
+    expect(screen.getByText('Hit the Enter key to create a new thought.')).toBeInTheDocument()
+
+    const user = userEvent.setup({ delay: null })
+    await act(() => user.keyboard('{Enter}'))
+    expect(screen.getByText('You did it!')).toBeInTheDocument()
+    expect(screen.getByText('Now type something. Anything will do.')).toBeInTheDocument()
+
+    await dispatch(newThought({ value: 'my first thought' }))
+    await act(vi.runOnlyPendingTimersAsync)
+    expect(screen.getByText('Click the Next button when you are ready to continue.')).toBeInTheDocument()
+  })
+
+  describe('step second thought - create another thought', () => {
+    it('prompts to type some text to the created thought, then congratulates on typing', async () => {
+      await dispatch([
+        newThought({ value: 'already created thought' }),
+        tutorialStep({ value: TUTORIAL_STEP_SECONDTHOUGHT }),
+      ])
+      await act(vi.runOnlyPendingTimersAsync)
+      expect(screen.getByText('Try adding another thought. Do you remember how to do it?')).toBeInTheDocument()
+
+      await dispatch(newThought({ value: 'another thought' }))
+      await act(vi.runOnlyPendingTimersAsync)
+      expect(screen.getByText('Good work!')).toBeInTheDocument()
+    })
+  })
+
+  it('step subthought - how to create a thought inside thought', async () => {
+    await dispatch([
+      tutorialStep({ value: TUTORIAL_STEP_SUBTHOUGHT }),
+      newThought({
+        value: 'parent',
+      }),
+      newThought({
+        value: 'child',
+        insertNewSubthought: true,
+      }),
+    ])
+    await act(vi.runOnlyPendingTimersAsync)
+
+    expect(screen.getByText(`As you can see, the new thought "child" is nested`, { exact: false })).toBeInTheDocument()
+    expect(getByText(screen.getByTestId('tutorial-step'), 'parent', { exact: false })).toBeInTheDocument()
+  })
+
+  describe('step autoexpand - show that thoughts expand and collapse on outside click', async () => {
+    beforeAll(() => dispatch(tutorialStep({ value: TUTORIAL_STEP_AUTOEXPAND })))
+    it('tells us about thoughts being hidden when clicked away', async () => {
+      expect(
+        screen.getByText('thoughts are automatically hidden when you click away', { exact: false }),
```

Comment:
Is there a reason to use `{ exact: false }` for all of these?

https://testing-library.com/docs/queries/about/#precision

Reply:
reason's simple -- a lot of times in the tutorial the text is quite long, spanning 3-4-5 lines
I tried to find the most crucial pieces of the text on a given step, that clearly indicates what's going on, without going through the pain of doing multi-line matches